pkg/config/config.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2021 Red Hat, Inc.
+// Copyright Contributors to the Open Cluster Management project
+package config
+
+import (
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/golang/glog"
+)
+
+const (
+	DEFAULT_SERVICE_PORT = ":3030"
+	DEFAULT_HTTP_TIMEOUT = 300000                                  // 5 min, to fix the EOF response at the collector
+	DEFAULT_USE_MOCK     = false                                   // Use Mocked Cluster ID ?
+	DEFAULT_CCX_SERVER   = "http://localhost:8080/api/v1/clusters" // For local use only
+)
+
+// Config - Define a config type to hold our config properties.
+type Config struct {
+	ServicePort string
+	HTTPTimeout int  // timeout when the http server should drop connections
+	UseMock     bool // Use Mock Server or actual endpoint
+	CCXServer   string
+	KubeConfig  string `env:"KUBECONFIG"` // Local kubeconfig path
+}
+
+// Cfg service configuration
+var Cfg = Config{}
+
+func init() {
+	// If environment variables are set, use those values constants
+	// Simply put, the order of preference is env -> default constants (from left to right)
+	setDefault(&Cfg.ServicePort, "SERVICE_PORT", DEFAULT_SERVICE_PORT)
+	setDefault(&Cfg.CCXServer, "CCX_SERVER", DEFAULT_CCX_SERVER)
+	setDefaultInt(&Cfg.HTTPTimeout, "HTTP_TIMEOUT", DEFAULT_HTTP_TIMEOUT)
+	setDefaultBool(&Cfg.UseMock, "USE_MOCK", DEFAULT_USE_MOCK)
+	defaultKubePath := filepath.Join(os.Getenv("HOME"), ".kube", "config")
+	if _, err := os.Stat(defaultKubePath); os.IsNotExist(err) {
+		// set default to empty string if path does not reslove
+		defaultKubePath = ""
+	}
+	setDefault(&Cfg.KubeConfig, "KUBECONFIG", defaultKubePath)
+
+}
+
+func setDefault(field *string, env, defaultVal string) {
+	if val := os.Getenv(env); val != "" {
+		glog.Infof("Using %s from environment: %s", env, val)
+		*field = val
+	} else if *field == "" && defaultVal != "" {
+		glog.Infof("%s not set, using default value: %s", env, defaultVal)
+		*field = defaultVal
+	}
+}
+
+// TODO: Combine with function above.
+func setDefaultInt(field *int, env string, defaultVal int) {
+	if val := os.Getenv(env); val != "" {
+		glog.Infof("Using %s from environment: %s", env, val)
+		var err error
+		*field, err = strconv.Atoi(val)
+		if err != nil {
+			glog.Error("Error parsing env [", env, "].  Expected an integer.  Original error: ", err)
+		}
+	} else if *field == 0 && defaultVal != 0 {
+		glog.Infof("No %s from file or environment, using default value: %d", env, defaultVal)
+		*field = defaultVal
+	}
+}
+
+func setDefaultBool(field *bool, env string, defaultVal bool) {
+	if val := os.Getenv(env); val != "" {
+		glog.Infof("Using %s from environment: %s", env, val)
+		var err error
+		*field, err = strconv.ParseBool(val)
+		if err != nil {
+			glog.Error("Error parsing env [", env, "].  Expected a boolean.  Original error: ", err)
+		}
+	} else {
+		glog.Infof("No %s from file or environment, using default value: %v", env, defaultVal)
+		*field = defaultVal
+	}
+}
```

Comment:
You could reuse tests from collector.
https://github.com/open-cluster-management/search-collector/blob/main/pkg/config/config_test.go